aten/src/TH/THTensorApply.h

```diff
@@ -319,43 +319,44 @@
 
 #define TH_TENSOR_APPLY_REDUCTION_OMP(TYPE, TENSOR, OPERATION, CODE) \
 {\
-  TYPE *rp = TENSOR->storage->data+TENSOR->storageOffset;                   \
   int TENSOR##Contg = THTensor_(isContiguous)(TENSOR);                      \
   ptrdiff_t TENSOR##Size = THTensor_(nElement)(TENSOR);                     \
-  ptrdiff_t iter = 0;                                                         \
   if(TENSOR##Contg){                                                         \
-    TYPE *TENSOR##_data = NULL;         \
-    PRAGMA( omp parallel for if (TENSOR##Size > TH_OMP_OVERHEAD_THRESHOLD_OMP) private(TENSOR##_data,  iter) reduction(OPERATION) ) \
+    ptrdiff_t iter = 0;                                                         \
+    TYPE *rp = TENSOR->storage->data+TENSOR->storageOffset;                   \
+    PRAGMA( omp parallel for if (TENSOR##Size > TH_OMP_OVERHEAD_THRESHOLD_OMP) firstprivate(rp) reduction(OPERATION) ) \
     for (iter = 0; iter < TENSOR##Size; iter++) { \
-      TENSOR##_data = rp+iter;                    \
+      TYPE *TENSOR##_data = rp+iter;                    \
       CODE                                         \
     }                                              \
   } else {                                         \
-    TH_UNUSED int TH_TENSOR_APPLY_hasFinished = 0;           \
+    int TH_TENSOR_APPLY_hasFinished = 0;           \
     int64_t TH_TENSOR_dim_index = 0;               \
     __TH_TENSOR_APPLYX_PREAMBLE(TYPE, TENSOR, -1, 1);\
-    PRAGMA(omp parallel if (TENSOR##Size > TH_OMP_OVERHEAD_THRESHOLD_OMP) firstprivate(TENSOR##_sizes, TENSOR##_strides, TENSOR##_dim, TENSOR##_stride, TENSOR##_size, TENSOR##_i) reduction(OPERATION))\
-    {\
-      size_t num_threads = omp_get_num_threads();\
-      size_t tid = omp_get_thread_num();\
-      size_t line_seg_length_avg = TENSOR##Size/num_threads;                                                     \
-      ptrdiff_t line_index_start = tid * line_seg_length_avg;                                            \
-      ptrdiff_t line_seg_length = (tid == num_threads - 1)? (TENSOR##Size - line_index_start):line_seg_length_avg;  \
-      __TH_TENSOR_APPLYX_CAL_MEMORY_OFFSET(TENSOR);\
-      TYPE *TENSOR##_data = rp + TENSOR##_memory_offset;\
-      ptrdiff_t count = 0;\
-      ptrdiff_t TENSOR##_start = TENSOR##_counter_tmp[TENSOR##_dim - 1];\
-      while(count < line_seg_length){\
-        for(TENSOR##_i=TENSOR##_start; (count < line_seg_length)&&(TENSOR##_i < TENSOR##_size); ++TENSOR##_i, ++count){\
-          CODE\
-          TENSOR##_data += TENSOR##_stride;\
-        }\
-        if(count < line_seg_length){\
-          __TH_TENSOR_APPLYX_UPDATE_COUNTERS_OMP(TENSOR);\
+    if (0 == TH_TENSOR_APPLY_hasFinished) {          \
```

Comment:
Why is this in a block now? Does this change the behavior? I'm not familiar with #2764 so this is probably a stupid question. :)

Reply:
It just follows the [previous usage](https://github.com/intel/pytorch/blob/c180e2106ad8c12329a5017010cf952a7a18483e/aten/src/TH/THTensorApply.h#L208). Actually, it's redundant check of size. If the size is zero,just skip the rest operation. It just follows the existed rules and remove the compilation warnings.

Reply:
I am sorry to introduce the aside effect when I committed the #2764.